File: Frontend/src/dto/fotodto.ts
import { UserDTO } from "./userdto";
import { AutoDTO } from "./auto";
export class FotoDTO {
    id: number;
    url: string;
    auto: AutoDTO;
    user: UserDTO;

    constructor (id: number,
                url: string,
                auto: AutoDTO,
                user: UserDTO) {
                    this.id = id;
                    this.url = url;
                    this.auto = auto;
                    this.user = user;
                }
}

File: Backend/carShop/src/main/java/com/contrader/carShop/dto/FotoDTO.java
package com.contrader.carShop.dto;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import com.contrader.carShop.model.Auto;
import com.contrader.carShop.model.User;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonIdentityInfo(generator= ObjectIdGenerators.PropertyGenerator.class, property="id")
public class FotoDTO {
    private Long id;
    private String url;

    private Auto auto;

    private User user;
}